### PR TITLE
AzureDbDataProvider - cache client

### PR DESCRIPTION
**Pull request by ovidiaconescu**
AzureDbDataProvider - GetClient() method first initializes the client and uses the cached one for further calls. This way the time spent to create the connection is only used once.

In a normal day-to-day scenario, this improves Audit InsertEvent performance dramatically. From my tests using the Azure CosmosDb Emulator, it's as fast as using the FileLogProvider.

### CODE REVIEW

File: src/Audit.NET.AzureDocumentDB/Providers/AzureDbDataProvider.cs [hunk 1]
```diff
@@ -110,32 +90,35 @@ public override T GetEvent<T>(object docId)
...
+        private void InitializeClient()
         {
-            return new DocumentClient(new Uri(_connectionString), _authKey);
+            var policy = new ConnectionPolicy
+            {
+                ConnectionMode = ConnectionMode.Direct,
+                ConnectionProtocol = Protocol.Tcp
+            };
+
+            client = new DocumentClient(new Uri(ConnectionString), AuthKey, policy);
```
Comment (thepirat000) [inline src/Audit.NET.AzureDocumentDB/Providers/AzureDbDataProvider.cs]: we should allow the user to passchange the policy

**Comment by thepirat000** [comment]
Nice!

I think we should also allow the client to pass the Policy and not hardcode that setting.

**Comment by ovidiaconescu** [comment]
Yes, that would be nice. I'm thinking that I'll also add the option to add the client, not the Connection String and AuthKey. As in my aspnetcore app I'm already instantiating it before in ConfigureServices and Audit gets created in Configure.

**Comment by ovidiaconescu** [comment]
Done.

**Comment by thepirat000** [comment]
Thanks! merging this soon

**Pull request merged**